### PR TITLE
fixed macOS build and added macOS and MSYS2 actions

### LINKED ISSUE #2: macos build error
Hi, I'm fixing macos build. I changed cmake file to solve VERSION file error and some compiler warning. But every time I get this error. Linux build succesfull btw. 

https://github.com/dsm/qucsator_rf/actions/runs/8970309257

```
[ 57%] Generating parse_citi.hpp, parse_citi.cpp
[ 58%] Generating scan_zvr.cpp
[ 59%] Generating parse_mdl.hpp, parse_mdl.cpp
[ 59%] Generating parse_touchstone.hpp, parse_touchstone.cpp
[ 59%] Generating parse_dataset.hpp, parse_dataset.cpp
[ 59%] Generating parse_zvr.hpp, parse_zvr.cpp
[ 60%] Generating scan_touchstone.cpp
[ 61%] Generating parse_csv.hpp, parse_csv.cpp
[ 61%] Generating scan_csv.cpp
[ 61%] Generating scan_dataset.cpp
[ 61%] Generating scan_citi.cpp
[ 63%] Generating scan_netlist.cpp
[ 62%] Generating scan_mdl.cpp
[ 63%] Generating parse_netlist.hpp, parse_netlist.cpp
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_dataset.ypp:52.14-23: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_dataset.hpp] Error 1
make[2]: *** Waiting for unfinished jobs....
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_touchstone.ypp:52.14-26: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_touchstone.hpp] Error 1
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_csv.ypp:52.14-19: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_csv.hpp] Error 1
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_netlist.ypp:49.14-23: syntax error, unexpected string, expecting =
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_mdl.ypp:52.14-19: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_netlist.hpp] Error 1
make[2]: *** [src/parse_mdl.hpp] Error 1
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_citi.ypp:52.14-20: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_citi.hpp] Error 1
/Users/runner/work/qucsator_rf/qucsator_rf/src/parse_zvr.ypp:52.14-19: syntax error, unexpected string, expecting =
make[2]: *** [src/parse_zvr.hpp] Error 1
make[1]: *** [src/CMakeFiles/libqucsator.dir/all] Error 2
make: *** [all] Error 2
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by dsm**
Hi, this PR made macos build succesfull.

-fixed VERSION file problem because of macos fs is case sensitive. 
-added macos github action.
-some cmake error fixes.

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -41,11 +41,11 @@ jobs:
...
           dos2unix 
           mingw-w64-ucrt-x86_64-gperf
-    - shell: msys2 {0}
+          
+    - name: Build with cmake
       run: |
-        set MSYSTEM=UCRT64
-        cmake.exe -B ${{github.workspace}}/build -G 'Unix Makefiles' -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
-        cmake.exe --build ${{github.workspace}}/build -j`nproc` --config ${{env.BUILD_TYPE}}
+        cmake -B ${{github.workspace}}/build -G 'Unix Makefiles' -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
```
Comment (ra3xdh) [inline .github/workflows/windows.yml]: Try using `-G "MinGW Makefiles"` option The Unix Makefiles generator may not work within MSYS2.

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -36,15 +36,17 @@ jobs:
           mingw-w64-ucrt-x86_64-qt6-tools 
           mingw-w64-ucrt-x86_64-qt6-svg 
           mingw-w64-ucrt-x86_64-make 
+          mingw-w64-ucrt-x86_64-ninja
```
Comment (ra3xdh) [inline .github/workflows/windows.yml:39]: Ninja will not work for Qucs-S

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -36,15 +36,17 @@ jobs:
...
           mingw-w64-ucrt-x86_64-python 
+          mingw-w64-ucrt-x86_64-autotools
```
Comment (ra3xdh) [inline .github/workflows/windows.yml]: Autotools is deprecated and not needed anymore.

**Comment by ra3xdh** [comment]
See my line comments on the commits related to `windows.yml` The  `-G "MinGW Makefiles"` option should help for MSYS2.

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -46,8 +46,9 @@ jobs:
           
     - name: Build with cmake
       run: |
-        cmake -B ${{github.workspace}}/build  -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
-        cmake --build ${{github.workspace}}/build -j`nproc` --config ${{env.BUILD_TYPE}}
+        ls -la
+        cmake.exe -B D:\a\qucsator_rf\qucsator_rf\build  -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
```
Comment (ra3xdh) [inline .github/workflows/windows.yml]: Cmake from MSYS2 requires explicit generator specification. Unspecified generator may not work.

**Comment by dsm** [comment]
in local msys2 -G 'Unix Makefiles' work in mingw64 and ucrt64 but i don't know why not working on ci

**Comment by dsm** [comment]
windows build using msys2 fixed and added basic github action.

**Comment by ra3xdh** [comment]
Thanks! Please squash 11 commits related to `windows.yml` using the Git interactive rebase `git rebase -i` and update your branch using force push. The I will merge this.

**Comment by dsm** [comment]
i fixed the 11 commits issue, i forgot to create different brach for this :)

**Pull request merged**